src/js/live-tracker.js

```diff
@@ -0,0 +1,204 @@
+import Component from './component.js';
+import mergeOptions from './utils/merge-options.js';
+
+/* track when we are at the live edge, and other helpers for live playback */
+class LiveTracker extends Component {
+
+  constructor(player, options) {
+    // LiveTracker does not need an element
+    const options_ = mergeOptions({createEl: false}, options);
+
+    super(player, options_);
+
+    this.on(this.player_, 'durationchange', this.handleDurationchange);
+  }
+
+  isBehind() {
+    const liveCurrentTime = this.liveCurrentTime();
+    const currentTime = this.player_.currentTime();
+    const segmentLength = this.segmentLength_ || 12;
+
+    // the live edge window is the amount of seconds away from live
+    // that a player can be, but still be considered live.
+    // we add 0.07 because the live tracking happens every 30ms
+    // and we want some wiggle room for short segment live playback
+    const liveEdgeWindow = segmentLength + 0.07;
+
+    return liveCurrentTime !== Infinity && (liveCurrentTime - liveEdgeWindow) >= currentTime;
```

Comment:
why would liveCurrentTime be `Infinity`?

Reply:
if seekable end is Infinity

Reply:
added